consumption.tex

```diff
@@ -0,0 +1,74 @@
+
+\subsection{Bootstraping Mechanism}
+\import{./}{algorithms/alg.burn-verifier.tex}
+
+Being able to verify events, we can finally make use of verified events in order to grant target cryptocurrency to users who burn source cryptocurrency. After burning on the source blockchain, the user needs to complete the event submission through \textsf{submit-event-proof} as described earlier. After the event is verified successfully and finalized, the user calls the \textsf{claim} function with the forementioned event. This function ensures that the event provided is finalized and has not been claimed before (i.e. noone has been granted target cryptocurrency for this specific event in the past). Then, after verifying through $\BurnVerify$ that the receiving address of the event is a burn address where the tag is the function caller's address, it releases the amount of coins burned in the form of an ERC-20 token. We present a contract with this capability in Algorithm~\ref{alg.burn-verifier} which we call the \textsf{burn-verifier}.
+
+In the interest of keeping this implementation generic we assume that the user receives a token in return for his burn. However, instead of minting a token, the target cryptocurrency could allow the burn verifier contract to mint native cryptocurrency for any user who successfully claims an event. This would allow the target cryptocurrency to be bootstrapped entirely though burning as desired.
+
+\subsection{Verifying Block Connection with a Federation}
+We now introduce a second way for proving and verifying the block connection.
+
+\import{./}{algorithms/alg.verify-event-federation.tex}
+
+Let us assume that the smart contract holds in its state the continuously updated set of all valid blocks on the best source blockchain, in the form of a Merkle Tree root $\mathcal{M}$. Then, by verifying a Merkle Tree inclusion proof \textsf{blockConnection} for $\textsf{b} \in \mathcal{M}$ a verifier can be certain that the block provided is indeed part of the best source blockchain. Combined with verifying the transaction this completes our $\textsf{verify}_\mathcal{M}^e$ function, shown in Algorithm~\ref{alg.verify-event-federation}.
+
+We now address the final issue of maintaining an up-to-date set of blocks in the form of $\mathcal{M}$. We propose a federation mechanism of source blockchain full nodes that monitor the source chain and vote for the new set of source chain blocks, if and when it changes. A majority of votes is necessary for $\mathcal{M}$ to be updated on the target blockchain. In this manner, the federation federation acts as a bridge from the source to the target blockchain. The security of our scheme relies on the assumption that the federation majority is honest.
```

Comment:
You should more clearly state what the federation is. It consists of a list of $2n$ public keys which are hard-coded in the smart contract, $n + 1$ of which are assumed to be controlled by honest parties.